quic/s2n-quic-core/src/stream/state.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::ensure;
+use core::fmt;
+
+pub type Result<T> = core::result::Result<(), Error<T>>;
+
+macro_rules! transition {
```

Comment:
should this be in a non-stream specific module? it could be used for some of the BBR enums that have state transitions